modules/kafka/kafka_test.go

```diff
@@ -74,6 +78,297 @@ func TestKafka_invalidVersion(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestKafka_networkConnectivity(t *testing.T) {
+	ctx := context.Background()
+	var err error
+
+	const (
+		// config
+		topic_in  = "topic_in"
+		topic_out = "topic_out"
+
+		address = "kafka:9092"
+
+		// test data
+		key      = "wow"
+		text_msg = "test-input-external"
+	)
+
+	Network, err := network.New(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// kafkaWithListener {
+	kafkaContainer, err := kafka.Run(ctx,
+		"confluentinc/confluent-local:7.6.1",
+		kafka.WithClusterID("test-cluster"),
+		network.WithNetwork([]string{"kafka"}, Network),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "BROKER",
+				Host: "kafka",
+				Port: "9092",
+			},
+		}),
+	)
+	// }
+	testcontainers.CleanupContainer(t, kafkaContainer)
+	require.NoError(t, err, "failed to start kafka container")
+
+	kcat, err := runKcatContainer(ctx, Network.Name, "/tmp/msgs.txt")
+	testcontainers.CleanupContainer(t, kcat)
+	require.NoError(t, err, "failed to start kcat")
+
+	// 4. Copy message to kcat
+	err = kcat.SaveFile(ctx, "Message produced by kcat")
+	require.NoError(t, err)
+
+	brokers, err := kafkaContainer.Brokers(context.TODO())
+	require.NoError(t, err, "failed to get brokers")
+
+	// err = createTopics(brokers, []string{topic_in, topic_out})
+	err = kcat.CreateTopic(ctx, address, topic_in)
+	require.NoError(t, err, "create topic topic_in")
+
+	err = kcat.CreateTopic(ctx, address, topic_out)
+	require.NoError(t, err, "create topic topic_out")
+
+	// perform assertions
+
+	// set config to true because successfully delivered messages will be returned on the Successes channel
+	config := sarama.NewConfig()
+	config.Producer.Return.Successes = true
+	config.Consumer.MaxWaitTime = 2 * time.Second
+
+	producer, err := sarama.NewSyncProducer(brokers, config)
+	require.NoError(t, err, "create kafka producer")
+
+	// Act
+
+	// External write
+	_, _, err = producer.SendMessage(&sarama.ProducerMessage{
+		Topic: topic_in,
+		Key:   sarama.StringEncoder(key),
+		Value: sarama.StringEncoder(text_msg),
+	})
+	require.NoError(t, err, "send message")
+
+	// Internal read
+	_, stdout, err := kcat.Exec(ctx, []string{"kcat", "-b", address, "-C", "-t", topic_in, "-c", "1"})
+	require.NoError(t, err)
+
+	out, err := io.ReadAll(stdout)
+	require.NoError(t, err, "read message in kcat")
+	require.Contains(t, string(out), text_msg)
+
+	// Internal write
+	tempfile := "/tmp/msgs.txt"
+
+	err = kcat.CopyToContainer(ctx, []byte(out), tempfile, 700)
+	require.NoError(t, err)
+
+	_, _, err = kcat.Exec(ctx, []string{"kcat", "-b", address, "-t", topic_out, "-P", "-l", tempfile})
+	require.NoError(t, err, "send message with kcat")
+
+	// External read
+	client, err := sarama.NewConsumerGroup(brokers, "groupName", config)
+	require.NoError(t, err, "create consumer group")
+
+	consumer, _, done, cancel := NewTestKafkaConsumer(t)
+
+	sCtx, _ := context.WithTimeout(context.Background(), time.Second)
```

Comment:
bug: as flagged by linter, need to cancel.

Reply:
I'm having issues with this test, see https://github.com/testcontainers/testcontainers-go/pull/2490#issuecomment-2493002440